app/helpers/admin/products_helper.rb

```diff
@@ -10,8 +10,11 @@ def product_image_form_path(product)
       end
     end
 
-    def prepare_new_variant(product)
-      product.variants.build
+    def prepare_new_variant(product, producer_options)
+      # e.g producer_options = [['producer name', id]]
+      product.variants.build do |new_variant|
+        new_variant.supplier_id = producer_options.first.second if producer_options.one?
```

Comment:
If only one producer exists, the producer column will be hidden according to the issue. In that case, the producer should be auto-assigned so that UI doesn't show an error on saving the variant.

Reply:
It would be nicer to have `producer_options` as an array of hashes:  `[ { name: 'producer name', id: id } ]`, anyway its fine to keep it like this.